src/applications/disability-benefits/all-claims/pages/reservesNationalGuardService.jsx

```diff
@@ -0,0 +1,163 @@
+    errors.addError('Please enter a valid 9 digit ZIP (dashes allowed)');
+  }
+}
+
+const updateStates = (form) => {
+  const currentCity = get(form, 'serviceInformation.reservesNationalGuardService.unitAddress.city', '').trim().toUpperCase();
+  if (MILITARY_CITIES.includes(currentCity)) {
+    return {
+      'enum': MILITARY_STATE_VALUES,
+      enumNames: MILITARY_STATE_LABELS
+    };
+  }
+
+  return {
+    'enum': STATE_VALUES,
+    enumNames: STATE_LABELS
+  };
+};
+const {
+  unitPhone,
+  unitAddress,
+  unitName,
+  obligationTermOfServiceDateRange
+} = fullSchema.properties.serviceInformation.properties.reservesNationalGuardService.properties;
+
+export const uiSchema = {
+  'ui:title': 'Reserves and National Guard Information',
+  'ui:description': ReservesGuardDescription,
+  serviceInformation: {
+    reservesNationalGuardService: {
+      obligationTermOfServiceDateRange: dateRangeUI(
+        'Obligation start date',
+        'Obligation end date',
+        'End date must be after start date'
+      ),
+      unitName: {
+        'ui:title': 'Unit name',
+      },
+      unitAddress: {
+        'ui:title': 'Unit address',
+        'ui:order': [
+          'country',
+          'addressLine1',
+          'addressLine2',
+          'addressLine3',
+          'city',
+          'state',
+          'zipCode'
+        ],
+        country: {
+          'ui:title': 'Country'
+        },
+        addressLine1: {
+          'ui:title': 'Street address'
+        },
+        addressLine2: {
+          'ui:title': 'Street address (optional)'
+        },
+        addressLine3: {
+          'ui:title': 'Street address (optional)'
+        },
+        city: {
+          'ui:title': 'City',
+          'ui:validations': [validateMilitaryCity]
+        },
+        state: {
+          'ui:title': 'State',
+          'ui:required': formData => get(formData, 'serviceInformation.reservesNationalGuardService.unitAddress.country', USA) === USA,
+          'ui:options': {
+            hideIf: formData => get(formData, 'serviceInformation.reservesNationalGuardService.unitAddress.country', USA) !== USA,
+            updateSchema: updateStates
+          },
+          'ui:validations': [{
+            validator: validateMilitaryState
+          }]
+        },
+        zipCode: {
+          'ui:title': 'ZIP code',
+          'ui:validations': [validateZIP],
+          'ui:required': formData => get(formData, 'serviceInformation.reservesNationalGuardService.unitAddress.country', USA) === USA,
+          'ui:errorMessages': {
+            pattern: 'Please enter a valid 5- or 9-digit ZIP code (dashes allowed)'
+          },
+          'ui:options': {
+            widgetClassNames: 'va-input-medium-large',
+            hideIf: formData => get(formData, 'serviceInformation.reservesNationalGuardService.unitAddress.country', USA) !== USA,
+          }
+        }
+      },
+      unitPhone: {
+        'ui:title': 'Unit phone number',
```

Comment:
I think in the past we've used `PhoneNumberWidget` to do fancy things like allow dashes.